File: Customer/src/main/java/com/postman/alt/service/CustomFieldService.java
package com.postman.alt.service;

import com.postman.alt.service.dto.CustomFieldCreateRequest;
import com.postman.alt.service.dto.CustomFieldResponse;
import com.postman.alt.service.dto.CustomFieldUpdateRequest;

import java.util.List;
import java.util.UUID;

public interface CustomFieldService {

    List<CustomFieldResponse> list(UUID projectId, UUID requesterId);

    CustomFieldResponse create(UUID projectId, UUID requesterId, CustomFieldCreateRequest request);

    CustomFieldResponse update(UUID projectId, UUID fieldId, UUID requesterId, CustomFieldUpdateRequest request);

    void delete(UUID projectId, UUID fieldId, UUID requesterId);
}

File: client/src/app/core/services/custom-field.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../config/api.config';
import { CustomFieldCreateRequest, CustomFieldResponse, CustomFieldUpdateRequest } from '../models/custom-field.model';

@Injectable({ providedIn: 'root' })
export class CustomFieldService {
  private readonly http = inject(HttpClient);

  private base(projectId: string): string {
    return `${API_BASE_URL}/api/projects/${projectId}/custom-fields`;
  }

  list(projectId: string): Observable<CustomFieldResponse[]> {
    return this.http.get<CustomFieldResponse[]>(this.base(projectId));
  }

  create(projectId: string, request: CustomFieldCreateRequest): Observable<CustomFieldResponse> {
    return this.http.post<CustomFieldResponse>(this.base(projectId), request);
  }

  update(projectId: string, fieldId: string, request: CustomFieldUpdateRequest): Observable<CustomFieldResponse> {
    return this.http.patch<CustomFieldResponse>(`${this.base(projectId)}/${fieldId}`, request);
  }

  delete(projectId: string, fieldId: string): Observable<void> {
    return this.http.delete<void>(`${this.base(projectId)}/${fieldId}`);
  }
}
